torch/csrc/jit/passes/onnx/preprocess_for_onnx.cpp

```diff
@@ -216,9 +216,84 @@ static void fuseListAndListUnpack(Block* b) {
   }
 }
 
+static void decomposeLinear(Block* b) {
```

Comment:
Is this tested anywhere?

Reply:
I was to hitting CI error on onnx export, where it complains `aten::linear` not in the onnx opset.
After adding the decompose pass, the test passed.

I can add a standalone cpp test to verify the decomposition.

Reply:
ended up doing a python test:
https://github.com/pytorch/pytorch/pull/50856/files#diff-7c9f7299daa27f03c538880940837c0007483c522c3efbee443148d309b6f9ebR337-R366

Reply:
I don't think we want the deccomposition on by default, but we might consider adding it in a more similar way to https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/passes/decompose_ops.cpp#L195. Additionally, if there aren't any other use cases for decomposing linear (and i cant think of any of the top of my head), it might be easier to have this implemented as an onnx decomposition cc @BowenBao

Reply:
This is inserted in `PreprocessForONNX` and exposed in python via `torch._C._jit_pass_onnx_preprocess`.
We didn't put it int `decompose_ops.cpp` because we don't want the op decomposition for fuser.

Reply:
Yea, I just mean the API for doing decomposition there is a little bit easier. I think this is fine for landing, someone from ONNX might want to rewrite it later.